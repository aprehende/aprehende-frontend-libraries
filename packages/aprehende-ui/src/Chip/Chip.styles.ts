import styled from '@emotion/styled';

export const ChipContainer = styled.div`
  width: fit-content;
  display: flex;
  align-items: center;
  padding: 0.4rem 0.5rem;
  border-radius: 16px;
  border: 1px solid #e0e0e0;
  & > img {
    margin-left: 0.2rem;
    width: 1.2rem;
    height: 1.2rem;
    cursor: pointer;
  }
`;
