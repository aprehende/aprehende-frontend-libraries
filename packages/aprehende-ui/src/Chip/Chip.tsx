import { PropsWithChildren } from 'react';

import { ChipContainer } from './Chip.styles';
import close from './close.svg';

interface ChipProps {
  onDelete?: () => void;
}

const Chip: React.FC<PropsWithChildren<ChipProps>> = ({
  children,
  onDelete,
}) => {
  return (
    <ChipContainer className="chip">
      {children}
      {onDelete && <img src={close} alt="close" onClick={onDelete} />}
    </ChipContainer>
  );
};

export default Chip;
