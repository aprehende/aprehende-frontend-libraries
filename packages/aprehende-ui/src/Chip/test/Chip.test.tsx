import { render } from '@testing-library/react';
import { describe, it, vi } from 'vitest';

import Chip from '../Chip';

describe('Chip', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Chip>Chip</Chip>);
    expect(container).toMatchSnapshot();
  });

  it('should not render close icon', () => {
    const { queryByAltText } = render(<Chip>Chip</Chip>);
    expect(queryByAltText('close')).not.toBeInTheDocument();
  });

  it('should render the chip with the text', () => {
    const { getByText } = render(<Chip>Chip</Chip>);
    expect(getByText('Chip')).toBeInTheDocument();
  });

  it('should render close icon', () => {
    const fn = vi.fn();
    const { getByAltText } = render(<Chip onDelete={fn}>Chip</Chip>);
    expect(getByAltText('close')).toBeInTheDocument();
  });

  it('should execute the onDelete function', () => {
    const fn = vi.fn();
    const { getByAltText } = render(<Chip onDelete={fn}>Chip</Chip>);
    getByAltText('close').click();
    expect(fn).toHaveBeenCalled();
  });
});
