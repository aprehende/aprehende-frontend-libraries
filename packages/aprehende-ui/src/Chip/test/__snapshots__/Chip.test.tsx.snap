// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip > should match the snapshot 1`] = `
<div>
  <div
    class="chip css-9lmvxa"
  >
    Chip
  </div>
</div>
`;
